dot/core/service.go

```diff
@@ -474,20 +475,17 @@ func (s *Service) handleChainReorg(prev, curr common.Hash) error {
 }
 
 func (s *Service) notifyRuntimeUpdated(version runtime.Version) {
-	s.runtimeChangedLock.RLock()
-	defer s.runtimeChangedLock.RUnlock()
+	s.runtimeUpdateSubscriptionsLock.RLock()
+	defer s.runtimeUpdateSubscriptionsLock.RUnlock()
 
-	if len(s.runtimeChanged) == 0 {
+	if len(s.runtimeUpdateSubscriptions) == 0 {
 		return
 	}
 
-	logger.Info("notifying runtime updated chans...", "chans", s.runtimeChanged)
-	for _, ch := range s.runtimeChanged {
+	logger.Info("notifying runtime updated chans...", "chans", s.runtimeUpdateSubscriptions)
+	for _, ch := range s.runtimeUpdateSubscriptions {
```

Comment:
it would be nice for this to use `sync.WaitGroup` to ensure that there are no blocked/leaked goroutines.  Right now this will just spawn `len(s.runtimeUpdateSubscriptions)`number of goroutines, that potentially could be blocked if there is no process reading from the channel.

Reply:
@timwu20  I've added the `sync.WaitGroup`, I'm not entirely sure I've done this the way you envision this, please take a look and let me know how I should adjust.

Reply:
It's not completely necessary to pass in the `*sync.WaitGroup`, `wg` can be accessed in the closure.